packages/semantic-tokens/src/components/link/tokens.ts

```diff
@@ -0,0 +1,43 @@
+import { tokens } from '@fluentui/tokens';
+import {
+  foregroundCtrlBrandHoverRaw,
+  foregroundCtrlBrandPressedRaw,
+  foregroundCtrlBrandRestRaw,
+  foregroundCtrlNeutralPrimaryRestRaw,
+  strokewidthDefaultRaw,
+} from '../../control/variables';
+import {
+  ctrlLinkForegroundBrandHoverRaw,
+  ctrlLinkForegroundBrandPressedRaw,
+  ctrlLinkForegroundBrandRestRaw,
+  ctrlLinkForegroundNeutralHoverRaw,
+  ctrlLinkForegroundNeutralPressedRaw,
+  ctrlLinkForegroundNeutralRestRaw,
+  ctrlLinkInlineShowunderlineatrestRaw,
+  ctrlLinkInlineStrokewidthHoverRaw,
+  ctrlLinkInlineStrokewidthRestRaw,
+  ctrlLinkInlineUnderlineDashedRaw,
+  ctrlLinkInlineUnderlineSolidFigmaonlyRaw,
+  ctrlLinkOnpageShowunderlineatrestRaw,
+  ctrlLinkOnpageStrokewidthHoverRaw,
+  ctrlLinkOnpageStrokewidthRestRaw,
+  ctrlLinkOnpageUnderlineDashedRaw,
+  ctrlLinkOnpageUnderlineSolidFigmaonlyRaw,
+} from './variables';
+export const ctrlLinkForegroundNeutralRest = `var(${ctrlLinkForegroundNeutralRestRaw}, var(${foregroundCtrlNeutralPrimaryRestRaw}, ${tokens.colorNeutralForeground2}))`;
+export const ctrlLinkInlineStrokewidthRest = `var(${ctrlLinkInlineStrokewidthRestRaw}, ${strokewidthDefaultRaw})`;
+export const ctrlLinkInlineStrokewidthHover = `var(${ctrlLinkInlineStrokewidthHoverRaw}, ${strokewidthDefaultRaw})`;
+export const ctrlLinkInlineUnderlineDashed = `var(${ctrlLinkInlineUnderlineDashedRaw})`;
+export const ctrlLinkInlineUnderlineSolidFigmaonly = `var(${ctrlLinkInlineUnderlineSolidFigmaonlyRaw})`;
+export const ctrlLinkForegroundNeutralHover = `var(${ctrlLinkForegroundNeutralHoverRaw}, var(${foregroundCtrlNeutralPrimaryRestRaw}, ${tokens.colorNeutralForeground2Hover}))`;
+export const ctrlLinkForegroundNeutralPressed = `var(${ctrlLinkForegroundNeutralPressedRaw}, var(${foregroundCtrlNeutralPrimaryRestRaw}, ${tokens.colorNeutralForeground2Pressed}))`;
+export const ctrlLinkForegroundBrandRest = `var(${ctrlLinkForegroundBrandRestRaw}, var(${foregroundCtrlBrandRestRaw}, ${tokens.colorBrandForegroundLink}))`;
+export const ctrlLinkForegroundBrandHover = `var(${ctrlLinkForegroundBrandHoverRaw}, var(${foregroundCtrlBrandHoverRaw}, ${tokens.colorBrandForegroundLinkHover}))`;
+export const ctrlLinkForegroundBrandPressed = `var(${ctrlLinkForegroundBrandPressedRaw}, var(${foregroundCtrlBrandPressedRaw}, ${tokens.colorBrandForegroundLinkPressed}))`;
+export const ctrlLinkOnpageStrokewidthRest = `var(${ctrlLinkOnpageStrokewidthRestRaw}, ${strokewidthDefaultRaw})`;
+export const ctrlLinkOnpageStrokewidthHover = `var(${ctrlLinkOnpageStrokewidthHoverRaw}, ${strokewidthDefaultRaw})`;
+export const ctrlLinkOnpageUnderlineDashed = `var(${ctrlLinkOnpageUnderlineDashedRaw})`;
+export const ctrlLinkOnpageUnderlineSolidFigmaonly = `var(${ctrlLinkOnpageUnderlineSolidFigmaonlyRaw})`;
+export const ctrlLinkInlineShowunderlineatrest = `var(${ctrlLinkInlineShowunderlineatrestRaw})`;
+export const ctrlLinkOnpageShowunderlineatrest = `var(${ctrlLinkOnpageShowunderlineatrestRaw})`;
```

Comment:
Using the `tokens` object is a potential blocker for Web Components because it means we need ALLLL the tokens just to use some. I don't know if it helps or hurts this conversation, but we do mirror and provide exports for the token var strings.

https://github.com/microsoft/fluentui/blob/master/packages/web-components/src/theme/design-tokens.ts

Reply:
Ok perfect we were looking at breaking these out as an optimization already but pulling it in via web-components should be better

Reply:
Following up on this - we might run into the same issue if we depend upon web-components - alternatively, we've talked about flat-exporting from @fluentui/tokens - would this be acceptable? Or is that dependency a hard blocker? If so our only option might be copying the file other to prevent any cross dependencies